lib/lspace.rb

```diff
@@ -79,6 +91,11 @@ def [](key)
   # @param [Object] key
   # @param [Object] value
   def []=(key, value)
+    if LSpace.current != self && self.parent
+      raise ArgumentError.new("You cannot modify a LSpace you are not currently inside of.")
```

Comment:
Hmm. What prevents me from doing, e.g. `LSpace.current.parent.enter { LSpace.current[:haha] = :busted }` and breaking the caching?